language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/BlockNodeContextProvider.java

```diff
@@ -136,9 +140,21 @@ public BlockNodeContextProvider(Class<T> attachmentPoint) {
         completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_RETRY.get()));
         completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_RETRY_TRANSACTION.get()));
         completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_MATCH.get()));
-        completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_RETURN.get()));
         completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_PANIC.get()));
         completionItems.add(new SnippetCompletionItem(context, Snippet.DEF_STREAM.get()));
+
+        Optional<SemanticModel> semanticModel = context.currentSemanticModel();
+        if (semanticModel.isPresent()) {
+            ReturnTypeFinder finder = new ReturnTypeFinder(semanticModel.get());
+            Optional<TypeSymbol> returnType = finder.getTypeSymbol(context.getNodeAtCursor());
+
+            if (returnType.isEmpty() || returnType.get().typeKind() == TypeDescKind.NIL) {
+                completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_RETURN_SC.get()));
+            } else {
+                completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_RETURN.get()));
+            }
+        }
```

Comment:
We can rewrite this section using Java streams:
```
        Optional<TypeSymbol> returnType = context.currentSemanticModel()
                .flatMap(semanticModel -> {
                    ReturnTypeFinder finder = new ReturnTypeFinder(semanticModel.get());
                    return finder.getTypeSymbol(context.getNodeAtCursor());
                });

        if (returnType.isEmpty() || returnType.get().typeKind() == TypeDescKind.NIL) {
            completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_RETURN_SC.get()));
        } else {
            completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_RETURN.get()));
        }
```

Reply:
Changed to
```
        Optional<TypeSymbol> returnType = context.currentSemanticModel()
                .flatMap(semanticModel -> {
                    ReturnTypeFinder finder = new ReturnTypeFinder(semanticModel);    //removed .get()
                    return finder.getTypeSymbol(context.getNodeAtCursor());
                });

        if (returnType.isEmpty() || returnType.get().typeKind() == TypeDescKind.NIL) {
            completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_RETURN_SC.get()));
        } else {
            completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_RETURN.get()));
        }
```